functional/adoptopenjdk/src/net/adoptopenjdk/test/AdoptOpenJDKVendorTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.adoptopenjdk.test;
+
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests whether "AdoptOpenJDK" appears as vendor in all the places it is supposed to.
+ */
+@Test(groups = {"level.extended"})
+public class AdoptOpenJDKVendorTest {
```

Comment:
As we are building AQA to be a set of tests that can be helpful to all vendors, this test could get updated to be VendorTest (or possibly even better, changed to be a PropertiesTest) that contains a set of test methods that checks the value of a java property is as expected for that vendor.

Reply:
You want me to define some data source that contains the properties for various vendors so that I can look it up by some key (like the vendor)? Do you want me to do that now?

Reply:
It's not pretty and we need some value from the outside to tell `VendorPropertiesTest` what vendor to expect. A simple map with hard coded values it not going to work except we want to maintain an endless list of values. And the test won't pass with AdoptOpenJDK because we don't set one value correctly.